### ISSUE TITLE
java.lang.NullPointerException: Access key ID cannot be blank.

### ISSUE THREAD

**Issue by raghavanrrs** [jenkinsci/aws-credentials-plugin#258]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
  Jenkins: 2.488
OS: Linux - 4.14.355-271.569.amzn2.x86_64
Java: 21.0.3 - Amazon.com Inc. (OpenJDK 64-Bit Server VM)
---
Parameterized-Remote-Trigger:3.2.0
additional-metrics:67.vcb_823d3999a_2
agent-maintenance:2.207.v73c521a_1d9db_
allure-jenkins-plugin:2.32.0
amazon-ecr:1.136.v914ea_5948634
amazon-inspector-image-scanner:373.v86d72c7b_5716
analysis-model-api:12.9.1
ansicolor:1.0.5
ant:511.v0a_a_1a_334f41b_
antisamy-markup-formatter:162.v0e6ec0fcfcf6
apache-httpcomponents-client-4-api:4.5.14-208.v438351942757
apache-httpcomponents-client-5-api:5.4-124.v31e2987e48f4
aqua-security-scanner:3.2.7
archived-artifact-url-viewer:1.1
artifact-manager-s3:894.v29efa_d1a_6383
artifact-promotion:0.5.2
artifactory:4.0.8
asm-api:9.7.1-97.v4cc844130d97
atlassian-bitbucket-server-integration:4.1.2
atlassian-jira-software-cloud:2.0.15
audit-log:1.3
audit-trail:382.vf64d6f626060
authentication-tokens:1.119.v50285141b_7e1
authorize-project:1.8.1
aws-credentials:238.v8fb_588a_2b_e67
aws-global-configuration:139.v0b_c2603876b_c
aws-java-sdk:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-api-gateway:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-autoscaling:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-cloudformation:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-cloudfront:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-codebuild:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-codedeploy:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-ec2:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-ecr:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-ecs:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-efs:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-elasticbeanstalk:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-elasticloadbalancingv2:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-iam:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-kinesis:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-lambda:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-logs:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-minimal:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-organizations:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-secretsmanager:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-sns:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-sqs:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk-ssm:1.12.772-474.v7f79a_2046a_fb_
aws-java-sdk2-core:2.29.34-8.v76857219b_b_92
aws-java-sdk2-ec2:2.29.34-8.v76857219b_b_92
aws-java-sdk2-sts:2.29.34-8.v76857219b_b_92
aws-lambda:0.5.10
aws-secrets-manager-credentials-provider:1.214.va_0a_d8268d068
aws-secrets-manager-secret-source:1.72.v61781b_35c542
badge:2.5
bitbucket:254.v9d8ec33a_7a_42
bitbucket-build-status-notifier:1.4.2
bitbucket-oauth:0.13
bitbucket-pullrequest-builder:1.5.0
bitbucket-push-and-pull-request:3.1.3
blueocean:1.27.16
blueocean-autofavorite:1.2.5
blueocean-bitbucket-pipeline:1.27.16
blueocean-commons:1.27.16
blueocean-config:1.27.16
blueocean-core-js:1.27.16
blueocean-dashboard:1.27.16
blueocean-display-url:2.4.3
blueocean-events:1.27.16
blueocean-git-pipeline:1.27.16
blueocean-github-pipeline:1.27.16
blueocean-i18n:1.27.16
blueocean-jwt:1.27.16
blueocean-personalization:1.27.16
blueocean-pipeline-api-impl:1.27.16
blueocean-pipeline-editor:1.27.16
blueocean-pipeline-scm-api:1.27.16
blueocean-rest:1.27.16
blueocean-rest-impl:1.27.16
blueocean-web:1.27.16
bootstrap5-api:5.3.3-1
bouncycastle-api:2.30.1.78.1-248.ve27176eb_46cb_
branch-api:2.1200.v4b_a_3da_2eb_db_4
build-environment:1.7
build-failure-analyzer:2.5.3
build-health-cache-updater:1.1
build-monitor-plugin:1.14-947.vfec2cf655fe2
build-name-setter:2.4.3
build-timeout:1.33
build-timestamp:1.0.4
build-user-vars-plugin:182.v378b_9f14b_487
build-view-column:0.3
build-with-parameters:76.v9382db_f78962
buildresult-trigger:0.18
built-on-column:1.4
caffeine-api:3.1.8-133.v17b_1ff2e0599
categorized-view:164.v1c1b_dd4cdb_62
changes-since-last-success:0.6
checks-api:2.2.1
cloud-stats:336.v788e4055508b_
cloudbees-bitbucket-branch-source:933.0.2
cloudbees-disk-usage-simple:232.v713eeed2e1f4
cloudbees-folder:6.975.v4161e479479f
cobertura:1.17
code-coverage-api:4.99.0
command-launcher:116.vd85919c54a_d6
commons-compress-api:1.26.1-2
commons-lang3-api:3.17.0-84.vb_b_938040b_078
commons-text-api:1.12.0-129.v99a_50df237f7
compact-columns:1.199.v61a_f51712072
conditional-buildstep:1.4.3
config-file-provider:980.v88956a_a_5d6a_d
configuration-as-code:1903.v004d55388f30
configuration-as-code-groovy:1.1
confluence-publisher:163.vf906edb_73cce
copyartifact:757.v05365583a_455
coverage:1.16.1
credentials:1393.v6017143c1763
credentials-binding:687.v619cb_15e923f
cron_column:1.7
custom-view-tabs:1.3
dark-theme:514.va_3ea_73d65dc1
dashboard-view:2.521.v339b_a_f4d8da_8
data-tables-api:2.1.8-1
delivery-pipeline-plugin:1.4.2
dependency-check-jenkins-plugin:5.6.0
dependency-track:5.2.0
depgraph-view:1.0.5
devops-portal:415.vaf57f86bb4d1
discard-old-build:1.07
display-url-api:2.209.v582ed814ff2f
docker-build-publish:1.4.0
docker-build-step:2.12
docker-commons:445.v6b_646c962a_94
docker-java-api:3.4.1-96.v77147a_de67f8
docker-plugin:1.7.0
docker-workflow:580.vc0c340686b_54
doclinks:0.7
durable-task:581.v299a_5609d767
ec2:1764.v71db_efb_46a_fe
ec2-fleet:3.2.0
echarts-api:5.5.1-4
eddsa-api:0.3.0-4.v84c6f0f4969e
email-ext:1861.vdb_d991590994
emailext-template:219.v14fff547f78d
embeddable-build-status:487.va_0ef04c898a_2
emoji-symbols-api:13.v723a_b_8e234d1
envinject:2.919.v009a_a_1067cd0
envinject-api:1.199.v3ce31253ed13
extended-read-permission:53.v6499940139e5
extensible-choice-parameter:1.8.1
external-monitor-job:215.v2e88e894db_f8
favorite:2.221.v19ca_666b_62f5
file-operations:331.v9d88a_0206e2e
flatpickr-api:4.6.13-5.v534d8025a_a_59
folder-auth:217.vd5b_18537403e
font-awesome-api:6.6.0-2
forensics-api:2.6.0
generic-webhook-trigger:2.2.5
git:5.6.0
git-changelog:3.41
git-client:6.1.0
git-parameter:0.10.0
git-server:126.v0d945d8d2b_39
github:1.40.0
github-api:1.321-478.vc9ce627ce001
github-autostatus:3.6.2
github-branch-source:1807.v50351eb_7dd13
gitlab-api:5.6.0-97.v6603a_83f8690
gitlab-oauth:1.20
gitlab-plugin:1.9.6
global-slack-notifier:1.5
gradle:2.14
greenballs:1.15.1
groovy:457.v99900cb_85593
groovy-postbuild:267.va_df06de9fa_fa_
gson-api:2.11.0-85.v1f4e87273c33
h2-api:11.1.4.199-30.v1c64e772f3a_c
handy-uri-templates-2-api:2.1.8-30.v7e777411b_148
hidden-parameter:374.v00e27777a_0eb_
htmlpublisher:1.37
http_request:1.19
hudson-pview-plugin:1.8
instance-identity:201.vd2a_b_5a_468a_a_6
ionicons-api:74.v93d5eb_813d5f
jackson2-api:2.17.0-379.v02de8ec9f64c
jacoco:3.3.7
jakarta-activation-api:2.1.3-1
jakarta-mail-api:2.1.3-1
javadoc:280.v050b_5c849f69
javax-activation-api:1.2.0-7
javax-mail-api:1.6.2-10
jaxb:2.3.9-1
jdk-tool:80.v8a_dee33ed6f0
jenkins-design-language:1.27.16
jenkinslint:0.14.0
jersey2-api:2.44-151.v6df377fff741
jjwt-api:0.11.5-112.ve82dfb_224b_a_d
job-dsl:1.90
job-import-plugin:3.6
jobConfigHistory:1283.veb_dfb_00b_5ec0
jobcacher:589.v10419b_45a_a_72
jobrequeue:1.1
joda-time-api:2.13.0-93.v9934da_29b_a_e9
jqs-monitoring:37.vf50a_82a_0b_f32
jquery:1.12.4-3
jquery3-api:3.7.1-2
jsch:0.2.16-86.v42e010d9484b_
json-api:20240303-101.v7a_8666713110
json-path-api:2.9.0-118.v7f23ed82a_8b_8
junit:1311.v39e1716e4eb_e
junit-attachments:239.v9e003a_c80a_8c
junit-realtime-test-reporter:149.v05a_d403e2f48
last-changes:2.8.0
ldap:770.vb_455e934581a_
leastload:62.vfa_8830902733
lockable-resources:1327.ved786b_a_197e0
log-parser:2.3.7
logging:1.0.0
mailer:489.vd4b_25144138f
mapdb-api:1.0.9-40.v58107308b_7a_7
mask-passwords:173.v6a_077a_291eb_5
matrix-auth:3.2.3
matrix-project:840.v812f627cb_578
maven-plugin:3.24
mercurial:1260.vdfb_723cdcc81
metrics:4.2.21-458.vcf496cb_839e4
mina-sshd-api-common:2.14.0-136.v4d2b_0853615e
mina-sshd-api-core:2.14.0-136.v4d2b_0853615e
monitoring:2.4.0
multiple-scms:0.8
nested-data-reporting:5.2.3
nested-view:1.36
next-executions:402.va_4407095f22a_
node-iterator-api:55.v3b_77d4032326
oidc-provider:79.v46f0066a_d813
okhttp-api:4.11.0-181.v1de5b_83857df
opentelemetry:3.1464.va_85780e90d4c
opentelemetry-api:1.44.1.40.v93f5f8ca_42c3
pam-auth:1.11
parameterized-scheduler:277.v61a_4b_a_49a_c5c
parameterized-trigger:806.vf6fff3e28c3e
performance:972.v0a_ed034885fd
periodicbackup:2.0
pipeline-aggregator-view:114.v98e9735f6193
pipeline-aws:1.45
pipeline-build-step:540.vb_e8849e1a_b_d8
pipeline-github-lib:61.v629f2cc41d83
pipeline-graph-analysis:216.vfd8b_ece330ca_
pipeline-graph-view:382.vb_9a_27b_7b_ea_71
pipeline-groovy-lib:744.v5b_556ee7c253
pipeline-input-step:495.ve9c153f6067b_
pipeline-maven:1332.v36ca_27d24383
pipeline-milestone-step:119.vdfdc43fc3b_9a_
pipeline-model-api:2.2218.v56d0cda_37c72
pipeline-model-definition:2.2218.v56d0cda_37c72
pipeline-model-extensions:2.2218.v56d0cda_37c72
pipeline-multibranch-defaults:2.1
pipeline-rest-api:2.34
pipeline-stage-step:312.v8cd10304c27a_
pipeline-stage-tags-metadata:2.2218.v56d0cda_37c72
pipeline-stage-view:2.34
pipeline-timeline:1.0.3
pipeline-utility-steps:2.18.0
plain-credentials:183.va_de8f1dd5a_2b_
plugin-util-api:5.1.0
popper-api:1.16.1-3
popper2-api:2.11.6-5
postbuild-task:1.9
postbuildscript:3.4.1-695.vf6b_0b_8053979
prism-api:1.29.0-18
progress-bar-column-plugin:11.vdef198c2d6c1
project-description-setter:1.2
project-health-report:1.2
project-stats-plugin:23.v47fee1f77b_84
pubsub-light:1.18
readonly-parameters:1.0.0
rebuild:332.va_1ee476d8f6d
resource-disposer:0.25
role-strategy:633.v836e5b_3e80a_5
run-condition:1.7
s3:483.vcb_db_3dcee68f
saferestart:0.7
saml:4.511.vef666858a_167
schedule-build:611.v23e5986ff480
scm-api:698.v8e3b_c788f0a_6
scoring-load-balancer:139.v1f21a_7494a_a_2
script-security:1369.v9b_98a_4e95b_2d
sectioned-view:1.29
sidebar-link:2.4.1
simple-theme-plugin:196.v96d9592f4efa_
slack:751.v2e44153c8fe1
snakeyaml-api:2.3-123.v13484c65210a_
solarized-theme:0.1
sonar:2.16.1
sonar-quality-gates:328.vf4369b_da_d3c2
sse-gateway:1.27
ssh-credentials:349.vb_8b_6b_9709f5b_
ssh-slaves:2.1010.v64ec48721231
ssh-steps:2.0.68.va_d21a_12a_6476
sshd:3.330.vc866a_8389b_58
stashNotifier:1.507.vb_7300a_1a_a_d10
structs:338.v848422169819
subversion:1281.vc8837f91a_07a_
systemloadaverage-monitor:1.2
testInProgress:1.4
testng-plugin:835.v51ed3da_fcc35
theme-manager:262.vc57ee4a_eda_5d
thinBackup:2.1.1
throttle-concurrents:2.16
timestamper:1.28
token-macro:400.v35420b_922dcb_
trilead-api:2.147.vb_73cc728a_32e
uno-choice:2.8.6
validating-string-parameter:251.vc34e592b_8a_4d
validating-yaml-parameter:259.v66e188d74dde
variant:60.v7290fc0eb_b_cd
view-job-filters:392.v2c0a_4dd46909
violation-comments-to-stash:1.134
warnings-ng:11.12.0
workflow-aggregator:600.vb_57cdd26fdd7
workflow-api:1336.vee415d95c521
workflow-basic-steps:1058.vcb_fc1e3a_21a_9
workflow-cps:4000.v5198556e9cea_
workflow-cps-global-lib:612.v55f2f80781ef
workflow-durable-task-step:1398.vf6c9e89e5988
workflow-job:1472.ve4d5eca_143c4
workflow-multibranch:795.ve0cb_1f45ca_9a_
workflow-scm-step:427.v4ca_6512e7df1
workflow-step-api:678.v3ee58b_469476
workflow-support:936.v9fa_77211ca_e1
ws-cleanup:0.48
zapper:1.0.7

  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Controller AL2
Agents AL2023

### Reproduction steps

1. Update aws-credentials plugin to version 238.v8fb_588a_2b_e67
2. Create AWS credentials with Global scope for AWS IAM role, leave Access Key ID and Secret Access Key blank, fill in IAM role only
3. run pipeline using the credentials created like below

agent {
      docker {
          image '<acc_id>.dkr.ecr.ap-southeast-1.amazonaws.com/iac-builder'
          registryUrl 'https://<acc_id>.dkr.ecr.ap-southeast-1.amazonaws.com'
          registryCredentialsId 'ecr:ap-southeast-1:jenkins-ecr'
          args "-v $HOME/.ssh:/home/jenkins/.ssh:ro -u jenkins:docker --privileged"
          reuseNode true
      }
  }

### Expected Results

[Pipeline] {
[Pipeline] withDockerRegistry
Using the existing docker config file.
Removing blacklisted property: auths
$ docker login -u AWS -p ******** https://<acc_id>.dkr.ecr.ap-southeast-1.amazonaws.com/
WARNING! Using --password via the CLI is insecure. Use --password-stdin.
WARNING! Your password will be stored unencrypted in /var/lib/jenkins/workspace/terraform-prod@tmp/23658396-8b55-4ab4-b371-9766c76d7215/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store

### Actual Results

Also:   org.jenkinsci.plugins.workflow.actions.ErrorAction$ErrorId: 6e48cdae-41ad-44d9-a10f-4e32bc2dd391
java.lang.NullPointerException: Access key ID cannot be blank.
	at PluginClassLoader for aws-java-sdk2-core//software.amazon.awssdk.utils.Validate.notNull(Validate.java:119)
	at PluginClassLoader for aws-java-sdk2-core//software.amazon.awssdk.auth.credentials.AwsBasicCredentials.<init>(AwsBasicCredentials.java:68)
	at PluginClassLoader for aws-java-sdk2-core//software.amazon.awssdk.auth.credentials.AwsBasicCredentials.<init>(AwsBasicCredentials.java:43)
	at PluginClassLoader for aws-java-sdk2-core//software.amazon.awssdk.auth.credentials.AwsBasicCredentials$Builder.build(AwsBasicCredentials.java:238)
	at PluginClassLoader for aws-java-sdk2-core//software.amazon.awssdk.auth.credentials.AwsBasicCredentials.create(AwsBasicCredentials.java:100)
	at PluginClassLoader for aws-credentials//com.cloudbees.jenkins.plugins.awscredentials.AWSCredentialsImpl.resolveCredentials(AWSCredentialsImpl.java:170)
	at PluginClassLoader for aws-credentials//com.cloudbees.jenkins.plugins.awscredentials.AWSCredentialsImpl.getCredentials(AWSCredentialsImpl.java:236)
	at PluginClassLoader for aws-java-sdk-minimal//com.amazonaws.http.AmazonHttpClient$RequestExecutor.getCredentialsFromContext(AmazonHttpClient.java:1295)
	at PluginClassLoader for aws-java-sdk-minimal//com.amazonaws.http.AmazonHttpClient$RequestExecutor.runBeforeRequestHandlers(AmazonHttpClient.java:869)
	at PluginClassLoader for aws-java-sdk-minimal//com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:818)
	at PluginClassLoader for aws-java-sdk-minimal//com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:805)
	at PluginClassLoader for aws-java-sdk-minimal//com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:779)
	at PluginClassLoader for aws-java-sdk-minimal//com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:735)
	at PluginClassLoader for aws-java-sdk-minimal//com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:717)
	at PluginClassLoader for aws-java-sdk-minimal//com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:581)
	at PluginClassLoader for aws-java-sdk-minimal//com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:559)
	at PluginClassLoader for aws-java-sdk-ecr//com.amazonaws.services.ecr.AmazonECRClient.doInvoke(AmazonECRClient.java:3770)
	at PluginClassLoader for aws-java-sdk-ecr//com.amazonaws.services.ecr.AmazonECRClient.invoke(AmazonECRClient.java:3737)
	at PluginClassLoader for aws-java-sdk-ecr//com.amazonaws.services.ecr.AmazonECRClient.invoke(AmazonECRClient.java:3726)
	at PluginClassLoader for aws-java-sdk-ecr//com.amazonaws.services.ecr.AmazonECRClient.executeGetAuthorizationToken(AmazonECRClient.java:1939)
	at PluginClassLoader for aws-java-sdk-ecr//com.amazonaws.services.ecr.AmazonECRClient.getAuthorizationToken(AmazonECRClient.java:1907)
	at PluginClassLoader for amazon-ecr//com.cloudbees.jenkins.plugins.amazonecr.AmazonECSRegistryCredential.getPassword(AmazonECSRegistryCredential.java:157)
	at PluginClassLoader for amazon-ecr//com.cloudbees.jenkins.plugins.amazonecr.AmazonECSRegistryTokenSource.convert(AmazonECSRegistryTokenSource.java:52)
	at PluginClassLoader for amazon-ecr//com.cloudbees.jenkins.plugins.amazonecr.AmazonECSRegistryTokenSource.convert(AmazonECSRegistryTokenSource.java:37)
	at PluginClassLoader for authentication-tokens//jenkins.authentication.tokens.api.AuthenticationTokens.convert(AuthenticationTokens.java:148)
	at PluginClassLoader for authentication-tokens//jenkins.authentication.tokens.api.AuthenticationTokens.convert(AuthenticationTokens.java:110)
	at PluginClassLoader for docker-commons//org.jenkinsci.plugins.docker.commons.credentials.DockerRegistryEndpoint.getToken(DockerRegistryEndpoint.java:237)
	at PluginClassLoader for docker-commons//org.jenkinsci.plugins.docker.commons.credentials.DockerRegistryEndpoint.newKeyMaterialFactory(DockerRegistryEndpoint.java:310)
	at PluginClassLoader for docker-workflow//org.jenkinsci.plugins.docker.workflow.RegistryEndpointStep$Execution2.newKeyMaterialFactory(RegistryEndpointStep.java:97)
	at PluginClassLoader for docker-workflow//org.jenkinsci.plugins.docker.workflow.AbstractEndpointStepExecution2.doStart(AbstractEndpointStepExecution2.java:52)
	at PluginClassLoader for workflow-step-api//org.jenkinsci.plugins.workflow.steps.GeneralNonBlockingStepExecution.lambda$run$0(GeneralNonBlockingStepExecution.java:77)
	at PluginClassLoader for opentelemetry-api//io.opentelemetry.context.Context.lambda$wrap$1(Context.java:241)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Finished: FAILURE

### Anything else?

earlier version working fine. Cannot rollback to previous versions.

**Comment by andrewhertog** [comment]
I'm also seeing the same issue, since upgrading to the latest version.

**Comment by Santiago8212** [comment]
I also have this error. I have corrected it by returning to the previous version, but jenkins auto updates it and the error persist.

**Comment by andrewhertog** [comment]
Still broken in the newest version 239

**Comment by basil** [comment]
Fixed in #260. Released in [240.v6d844a_6f5480](https://github.com/jenkinsci/aws-credentials-plugin/releases/tag/240.v6d844a_6f5480).
